src/Components/AffectedClustersTable/AffectedClustersTable.js

```diff
@@ -0,0 +1,257 @@
+    const newFilteredRows = buildFilteredRows(rows, newActiveFilters);
+    const newDisplayedRows = buildDisplayedRows(newFilteredRows, 1, perPage);
+    const newActiveChips = updateNameChip(activeChips, value);
+    setPage(1);
+    setActiveChips(newActiveChips);
+    setActiveFilters(newActiveFilters);
+    setFilteredRows(newFilteredRows);
+    setDisplayedRows(newDisplayedRows);
+  };
+
+  const filterConfig = {
+    items: [
+      {
+        label: 'Name',
+        type: conditionalFilterType.text,
+        filterValues: {
+          key: 'name-filter',
+          onChange: (_e, value) => onNameFilterChange(value),
+          value: activeFilters.name,
+        },
+      },
+    ],
+    isDisabled: isError || (data?.data && data?.data.length === 0),
+  };
+
+  const onSort = (_e, index, direction) => {
+    setActiveSortIndex(index);
+    setActiveSortDirection(direction);
+    // sorts the rows
+    const updatedRows = filteredRows.concat().sort((a, b) => {
+      if (direction === 'asc') {
+        return a.localeCompare(b);
+      }
+      return b.localeCompare(a);
+    });
+    setFilteredRows(updatedRows);
+    setDisplayedRows(buildDisplayedRows(updatedRows, page, perPage));
+  };
+
+  const onSetPage = (_e, newPage) => {
+    setDisplayedRows(buildDisplayedRows(filteredRows, newPage, perPage));
+    setPage(newPage);
+  };
+
+  const onSetPerPage = (_e, newPerPage) => {
+    setDisplayedRows(buildDisplayedRows(filteredRows, 1, newPerPage));
+    setPage(1);
+    setPerPage(newPerPage);
+  };
+
+  // constructs array of rows (from the initial data) checking currently applied filters
+  const buildFilteredRows = (allRows, filters) => {
+    const rows = allRows;
+    return rows.filter((row) => {
+      return Object.entries(filters).every((filter) => {
+        if (filter[0] === 'name') {
+          const filterNameValue = filter[1];
+          return row.includes(filterNameValue);
+        }
+        return false;
+        // further filters will be added soon
+      });
+    });
+  };
+
+  const buildDisplayedRows = (rows, page, perPage) => {
+    return rows.slice(perPage * (page - 1), perPage * (page - 1) + perPage);
+  };
+
+  useEffect(() => {
+    const newFilteredRows = buildFilteredRows(rows, activeFilters);
+    const newDisplayedRows = buildDisplayedRows(newFilteredRows, page, perPage);
+    setFilteredRows(newFilteredRows);
+    setDisplayedRows(newDisplayedRows);
+  }, [affectedClusters]);
+
+  return (
+    <React.Fragment>
+      <PrimaryToolbar
+        filterConfig={filterConfig}
+        pagination={{
+          itemCount: filteredRows.length,
+          page,
+          perPage,
+          onSetPage: onSetPage,
+          onPerPageSelect: onSetPerPage,
+        }}
+        activeFiltersConfig={{
+          filters: activeChips,
+          onDelete: onChipDelete,
+        }}
+      />
+      {(isUninitialized || isLoading || isFetching) && <Loading />}
```

Comment:
SO i thought about this (when implementing rtkquery) isloading kinda speaks to the component? is only false once... isFetching happens each time the input changes including that first time... think yah could kill the isLoading here maybe? 🤔